### ISSUE TITLE
Add support for instruction caches

### ISSUE THREAD

**Issue by CapdinCrando** [jhunterh/Cache-Hierarchy-Simulation-Tool#16]

**Comment by jhunterh** [comment]
not sure about this one. I'm pretty sure its possible to get the instruction pointer using PIN, but not sure how much this might increase the scope of the simulator. maybe a stretch goal?

**Comment by CapdinCrando** [comment]
Yeah, probably a stretch goal, but my thought was that since instructions are also stored in L3 Cache, then it could affect the hit/miss rate of a system with an L3 cache.
